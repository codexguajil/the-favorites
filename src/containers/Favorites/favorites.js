import React, { Component } from 'react';
import Movie from '../Movie/Movie';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import propTypes from 'prop-types';

class Favorites extends Component{
  filterMovies = () => {
    let favoritesArray = [];
    this.props.favorites.forEach(favorite => {
       this.props.movies.forEach(movie => {
         if(movie.id === favorite) {
           favoritesArray.push(movie)
         }
      })
    })
    return this.renderMovies(favoritesArray)
  }

  renderMovies = (movies) => {
    return movies.map(movie => {
      return (
        <Movie  
          {...movie}
        />
      )
    })
  }

  render() {
    return (
      <div className="movie-container">
        {this.filterMovies()}
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  movies: state.movies,
  favorites: state.favorites
})

export default connect(mapStateToProps)(Favorites)

Favorites.propTypes = {
  movies: propTypes.array.isRequired,
  favorites: propTypes.array.isRequired
}