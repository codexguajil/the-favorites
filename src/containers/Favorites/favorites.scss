.favorites-container {
  display: inline-flex;
  margin: auto;
}

.back-btn-fave {
  border-radius: 5px;
  color: hsla(0, 0%, 10%, 1);
  display: block;
  font-size: 22px;
  font-weight: bolder;
  height: 50px;
  margin: auto;
  margin-top: 18px;
  margin-bottom: 50px;
  width: 200px;
  background: hsla(0, 0%, 90%, 1);
  &:hover {
    background: hsla(0, 0%, 10%, 1);
    border: none;
    color: hsla(360, 0%, 100%, 1);
  }
}