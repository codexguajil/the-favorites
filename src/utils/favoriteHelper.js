export const  validateFavorite = async (movie) => {
    const { user_id, id } = this.props
    const url = `http://localhost:3000/api/users/${user_id}/favorites`      
    if(user_id) {
      const allFavorites = await fetchData(url)
      allFavorites.data.forEach(favorite => {
        if(favorite.movie_id === id) {
          console.log('alreay in favorites')
        } else {
          this.addFavorite()
        }
      })
    } else {
      this.setState()
    }
  }

export const  addFavorite = async (e) => {
    const { id, title, rating, user_id, posterImage, synopsis, releaseDate } = this.props
    const url = "http://localhost:3000/api/users/favorites/new"
    const movie = {
      movie_id: id, 
      user_id: user_id, 
      title: title, 
      poster_path: posterImage, 
      release_date: releaseDate, 
      vote_average: rating, 
      overview: synopsis
    }
    try {
      await fetchData(url, {
        method: "POST", 
        body: JSON.stringify(movie),
        headers: {
          "Content-Type": "application/json"
        }
      })
    } catch(error) {
      console.log(error.message)
    }
  }