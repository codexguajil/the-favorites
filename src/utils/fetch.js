export const fetchData = async (url, options) => {
  const response = await fetch(url, options)
  if(response.ok) {
    const parsedData = response.json()
    return parsedData;
  } else {
    return 'error'
  }
} 