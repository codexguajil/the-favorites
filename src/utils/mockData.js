export const mockUser = {
  id: 1,
  name: "Taylor",
  password: "password",
  email: "tman2272@aol.com"
}

export const mockMovieResponse = {
page: 1,
results: [
  {adult: false,
  backdrop_path: "/hMANgfPHR1tRObNp2oPiOi9mMlz.jpg",
  genre_ids: [28, 12, 878],
  id: 424783,
  original_language: "en",
  original_title: "Bumblebee",
  overview: "On the run in the year 1987, Bumblebee finds refuge in a junkyard in a small Californian beach town. Charlie, on the cusp of turning 18 and trying to find her place in the world, discovers Bumblebee, battle-scarred and broken.  When Charlie revives him, she quickly learns this is no ordinary yellow VW bug.",
  popularity: 93.533,
  poster_path: "/fw02ONlDhrYjTSZV8XO6hhU3ds3.jpg",
  release_date: "2018-12-15",
  title: "Bumblebee",
  video: false,
  vote_average: 6.5,
  vote_count: 1414},
  {adult: false,
  backdrop_path: "/9QusGjxcYvfPD1THg6oW3RLeNn7.jpg",
  genre_ids:[28, 12, 14, 878],
  id: 297802,
  original_language: "en",
  original_title: "Aquaman",
  overview: "Once home to the most advanced civilization on Earth, Atlantis is now an underwater kingdom ruled by the power-hungry King Orm. With a vast army at his disposal, Orm plans to conquer the remaining oceanic people and then the surface world. Standing in his way is Arthur Curry, Orm's half-human, half-Atlantean brother and true heir to the throne.",
  popularity: 165.687,
  poster_path: "/5Kg76ldv7VxeX9YlcQXiowHgdX6.jpg",
  release_date: "2018-12-07",
  title: "Aquaman",
  video: false,
  vote_average: 6.8,
  vote_count: 4652,
  },
  {adult: false,
  backdrop_path: "/bTeRgkAavyw1eCtSkaww18wLYNP.jpg",
  genre_ids:[80, 18, 53],
  id: 504172,
  original_language: "en",
  original_title: "The Mule",
  overview: "Earl Stone, a man in his 80s who is broke, alone, and facing foreclosure of his business when he is offered a job that simply requires him to drive. Easy enough, but, unbeknownst to Earl, he’s just signed on as a drug courier for a Mexican cartel. He does so well that his cargo increases exponentially, and Earl hit the radar of hard-charging DEA agent Colin Bates.",
  popularity: 128.205,
  poster_path: "/oeZh7yEz3PMnZLgBPhrafFHRbVz.jpg",
  release_date: "2018-12-14",
  title: "The Mule",
  video: false,
  vote_average: 6.5,
  vote_count: 876},
  {adult: false,
  backdrop_path: "/uUiId6cG32JSRI6RyBQSvQtLjz2.jpg",
  genre_ids:[28, 12, 16, 878, 35],
  id: 324857,
  original_language: "en",
  original_title: "Spider-Man: Into the Spider-Verse",
  overview: "Miles Morales is juggling his life between being a high school student and being a spider-man. When Wilson Kingpin Fisk uses a super collider, others from across the Spider-Verse are transported to this dimension.",
  popularity: 157.433,
  poster_path: "/iiZZdoQBEYBv6id8su7ImL0oCbD.jpg",
  release_date: "2018-12-07",
  title: "Spider-Man: Into the Spider-Verse",
  video: false,
  vote_average: 8.5,
  vote_count: 2715},
  {backdrop_path: "/e7VzDMrYKXVrVon04Uqsrcgnf1k.jpg",
  first_air_date: "2019-03-15",
  genre_ids: [16, 10765],
  id: 86831,
  name: "Love, Death & Robots",
  origin_country: [],
  original_language: "en",
  original_name: "Love, Death & Robots",
  overview: "Terrifying creatures, wicked surprises and dark comedy converge in this NSFW anthology of animated stories presented by Tim Miller and David Fincher.",
  popularity: 161.862,
  poster_path: "/rlrRI2b6RvM9I9xOBTKqcTaehkE.jpg",
  vote_average: 8.6,
  vote_count: 29}],
  total_pages: 1000,
  total_results: 20000
}


export const mockDataResponse = [
  {id: 424783,
  posterImage: "https://image.tmdb.org/t/p/w500/fw02ONlDhrYjTSZV8XO6hhU3ds3.jpg",
  rating: 6.5,
  releaseDate: "2018-12-15",
  synopsis: "On the run in the year 1987, Bumblebee finds refuge in a junkyard in a small Californian beach town. Charlie, on the cusp of turning 18 and trying to find her place in the world, discovers Bumblebee, battle-scarred and broken.  When Charlie revives him, she quickly learns this is no ordinary yellow VW bug.",
  title: "Bumblebee"},
  {id: 297802,
  posterImage: "https://image.tmdb.org/t/p/w500/5Kg76ldv7VxeX9YlcQXiowHgdX6.jpg",
  rating: 6.8,
  releaseDate: "2018-12-07",
  synopsis: "Once home to the most advanced civilization on Earth, Atlantis is now an underwater kingdom ruled by the power-hungry King Orm. With a vast army at his disposal, Orm plans to conquer the remaining oceanic people and then the surface world. Standing in his way is Arthur Curry, Orm's half-human, half-Atlantean brother and true heir to the throne.",
  title: "Aquaman"},
  {id: 504172,
  posterImage: "https://image.tmdb.org/t/p/w500/oeZh7yEz3PMnZLgBPhrafFHRbVz.jpg",
  rating: 6.5,
  releaseDate: "2018-12-14",
  synopsis: "Earl Stone, a man in his 80s who is broke, alone, and facing foreclosure of his business when he is offered a job that simply requires him to drive. Easy enough, but, unbeknownst to Earl, he’s just signed on as a drug courier for a Mexican cartel. He does so well that his cargo increases exponentially, and Earl hit the radar of hard-charging DEA agent Colin Bates.",
  title: "The Mule"},
  {id: 399361,
  posterImage: "https://image.tmdb.org/t/p/w500/aBw8zYuAljVM1FeK5bZKITPH8ZD.jpg",
  rating: 6.3,
  releaseDate: "2019-03-06",
  synopsis: "Struggling to make ends meet, former special ops soldiers reunite for a high-stakes heist: stealing $75 million from a South American drug lord.",
  title: "Triple Frontier"},
  {id: 86831,
  posterImage: "https://image.tmdb.org/t/p/w500/rlrRI2b6RvM9I9xOBTKqcTaehkE.jpg",
  rating: 8.6,
  releaseDate: undefined,
  synopsis: "Terrifying creatures, wicked surprises and dark comedy converge in this NSFW anthology of animated stories presented by Tim Miller and David Fincher.",
  title: "Love, Death & Robots"}
]