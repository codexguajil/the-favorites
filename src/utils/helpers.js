import { fetchData } from './fetch';

export const cleanMovieData = (movies) => {
  return movies.results.map(movie => {
    return ({
        title: movie.title || movie.name,
        rating: movie.vote_average,
        id: movie.id,
        releaseDate: movie.release_date,
        synopsis: movie.overview,
        posterImage: movie.poster_path
    })
  })
}

export const fetchAllFavorites = async (id) => {
  const url = `http://localhost:3000/api/users/${id}/favorites`;
  const fetchedFavorites = await fetchData(url);
  const cleanedFavorites = cleanFavorites(fetchedFavorites.data)
  return cleanedFavorites;
}

const cleanFavorites = (favorites) => {
  return favorites.map(movie => {
    return movie.movie_id
  })
}