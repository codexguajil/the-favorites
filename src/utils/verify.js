import { fetchData } from './fetch';

export const verifyUser = (data, user) => {
  data.forEach(dataUser => {
    if(dataUser.email === user.email && dataUser.password === user.password) {
      console.log('match!')
    } else if(dataUser.email === user.email && dataUser.password !== user.password) {
      console.log('wrong password!')
    } else {
      console.log('user doesnt exist')
    }
  })
}

