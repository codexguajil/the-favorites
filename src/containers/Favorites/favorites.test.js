import React from 'react';
import Favorites from './favorites'
import { shallow } from 'enzyme'
import { mockDataResponse, mockFavorite } from '../../utils/mockData'

describe('favorites', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      favorites: [1, 2, 3, 4],
      movies: mockDataResponse
    }

    wrapper = shallow(
      <Favorites {...props}/>
    )
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  describe('filterMovies', () => {
    it('should call renderMovies', () => {
      wrapper.instance().renderMovies = jest.fn
      // const mockFavorites = mockFavorite
      wrapper.instance().filterMovies()

      expect(wrapper.instance().renderMovies).toHaveBeenCalled()
    })
  })
  


})