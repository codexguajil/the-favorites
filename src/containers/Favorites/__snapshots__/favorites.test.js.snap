// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`favorites should match snapshot with favorites 1`] = `
<div>
  <Link
    className="back-btn-fave"
    to="/"
  >
    Back To Home
  </Link>
  <div
    className="movie-container"
  >
    <h2>
      You currently have no favorites saved.
    </h2>
  </div>
</div>
`;

exports[`favorites should match snapshot without favorites 1`] = `
<div>
  <Link
    className="back-btn-fave"
    to="/"
  >
    Back To Home
  </Link>
  <div
    className="movie-container"
  >
    <h2>
      You currently have no favorites saved.
    </h2>
  </div>
</div>
`;
